dio/lib/src/transformer.dart

```diff
@@ -138,27 +140,26 @@ class DefaultTransformer extends Transformer {
 
     if (options.responseType == ResponseType.bytes) return responseBytes;
 
-    String responseBody;
+    String? responseBody;
     if (options.responseDecoder != null) {
-      responseBody = options.responseDecoder(
-          responseBytes, options, response..stream = null);
+      responseBody = options.responseDecoder!(
+          responseBytes, options, response..stream = Stream.empty());
     } else {
       responseBody = utf8.decode(responseBytes, allowMalformed: true);
     }
-    if (responseBody != null &&
-        responseBody.isNotEmpty &&
+    if (responseBody.isNotEmpty &&
         options.responseType == ResponseType.json &&
         _isJsonMime(response.headers[Headers.contentTypeHeader]?.first)) {
       if (jsonDecodeCallback != null) {
-        return jsonDecodeCallback(responseBody);
+        return jsonDecodeCallback!(responseBody);
```

Comment:
Is this conversion needed?

Reply:
Don't think it should be necessary since we're doing a null check already, no?

Reply:
Yes. I think VS tooling hasn't caught up yet for proper null safety checks. At the time of writing, I couldn't get the IDE to not complain here

Reply:
@harsimranmaan I think this a limitation of the language and nothing to do with VS tooling. You will have to declare a local variable like `final decodeCallback = jsonDecodeCallback` and use that.

See here for more information: https://stackoverflow.com/questions/65456958/dart-null-safety-doesnt-work-with-class-fields

Reply:
Yes, we currently cannot promote on class members. Using a local variable is a good workaround.